timelock-impl/src/main/java/com/palantir/atlasdb/timelock/lock/watch/LockWatchTestingService.java

```diff
@@ -0,0 +1,137 @@
+import com.palantir.lock.watch.LockWatchStateUpdate;
+import com.palantir.lock.watch.UnlockEvent;
+import com.palantir.logsafe.SafeArg;
+
+public class LockWatchTestingService {
+    private final Logger log = LoggerFactory.getLogger(LockWatchTestingService.class);
+    private final ScheduledExecutorService executor = PTExecutors.newSingleThreadScheduledExecutor();
+    private final Supplier<LockWatchTestRuntimeConfig> runtime;
+    private final Function<String, LockWatchingResource> resource;
+    private final OpenLocksFilter filter = new OpenLocksFilter();
+
+    private LockWatchTestingService(
+            Supplier<LockWatchTestRuntimeConfig> runtime,
+            Function<String, LockWatchingResource> resource) {
+        this.runtime = runtime;
+        this.resource = resource;
+    }
+
+    public static void create(
+            Supplier<LockWatchTestRuntimeConfig> runtime,
+            Function<String, LockWatchingResource> resource) {
+        LockWatchTestingService testService = new LockWatchTestingService(runtime, resource);
+        testService.start();
+    }
+
+    private void start() {
+        executor.scheduleAtFixedRate(this::runOneIteration, 5, 5, TimeUnit.MINUTES);
+    }
+
+    private void runOneIteration() {
+        try {
+            LockWatchTestRuntimeConfig config = runtime.get();
+            if (config.namespaceToWatch().isPresent()) {
+                LockWatchingResource lockWatcher = resource.apply(config.namespaceToWatch().get());
+                LockWatchRequest request = LockWatchRequest.of(config.tablesToWatch().stream()
+                        .map(TableReference::getQualifiedName)
+                        .map(LockWatchReferences::entireTable)
+                        .collect(Collectors.toSet()));
+                LockWatchStateUpdate versionBefore = lockWatcher.getWatchState(
+                        OptionalLong.empty());
+
+                long startRegistering = System.currentTimeMillis();
+                lockWatcher.startWatching(request);
+                log.info("Registered lock watches for keyspace {} and tables {} in {} seconds.",
+                        SafeArg.of("keyspace", config.namespaceToWatch().get()),
+                        SafeArg.of("tables", config.tablesToWatch()),
+                        SafeArg.of("seconds", System.currentTimeMillis() - startRegistering));
+
+                long startUpdate = System.currentTimeMillis();
+                LockWatchStateUpdate versionAfter = lockWatcher.getWatchState(versionBefore.lastKnownVersion());
+                long duration = System.currentTimeMillis() - startUpdate;
+                if (versionAfter.success()) {
+                    Optional<LockWatchEvent> result = versionAfter.events().stream()
+                            .map(event -> event.accept(filter))
+                            .filter(Optional::isPresent)
+                            .map(Optional::get)
+                            .findFirst();
```

Comment:
We probably want to accumulate matching results since multiple nodes could be doing the registration.